### PR TITLE
Fix llvm, update sln/proj to vs2015, update to llvm 3.6, improve llvm recompilation

### PR DESCRIPTION
**Pull request by vlj**
This PR does 4 things:
- It fixes llvm with current master, thanks to fix I get from Neko
- It updates sln and project file to use VS2015, whose community version was released yesterday (and is required for dx12)
- It updates llvm to 3.6 version. llvm 3.5 doesn't build with VS2015. On the other hand, llvm 3.6 builds with VS 2013 but there are some issues when building rpcs3 with llvm support with VS2013 that doesn't appear with VS2015. While maybe we could fix VS2013 and llvm 3.6, it's easier to just use VS 2015. Plus it's needed for DX12.
- It simplifies LLVMRecompiler code, fix some data race condition and allow LLVM compiled block to be freed when they're replaced with another version (previously jitted code was not cleant, it worked since jit has a lower memory footprint than MCJIT but LLVM 3.6 removed JIT support in favor of the latter).

There are still some issues, exiting a game leads to a crash in a CHECK_STATUS support and I don't understand why.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,17 +1,18 @@
 [submodule "wxWidgets"]
 	path = wxWidgets
-	url = https://github.com/RPCS3/wxWidgets.git
+	url = https://github.com/Zangetsu38/wxWidgets
 	ignore = dirty
```
Comment (Nekotekina) [inline .gitmodules]: Let's modify RPCS3/wxWidgets instead.

Comment (Bigpet) [inline .gitmodules]: Alright I send a pull-request upstream https://github.com/wxWidgets/wxWidgets/pull/48 once that is merged I will rebase our little silly "add setup.h" change on top of that in RPCS3/wxWidgets

File: .gitmodules [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+	branch = master
 [submodule "rpcs3-ffmpeg"]
 	path = ffmpeg
-	url = https://github.com/hrydgard/ppsspp-ffmpeg
+	url = https://github.com/bigpet/ppsspp-ffmpeg
```
Comment (Bigpet) [inline .gitmodules]: same for this one, let me talk to hrygard again and I'll see if we can't work out a way to get VS2015 support in his repo. There's really no reason to fork it yet.

**Comment by vlj** [comment]
I fixed Release non llvm mode (and likely debug non llvm mode too)

**Comment by Nekotekina** [comment]
I removed some changes from my PR but seems it conflicted anyway, sorry.

**Comment by tambry** [comment]
Looks like the cmake file also needs to be updated to compile with llvm 3.6.

**Comment by Zangetsu38** [comment]
@Bigpet There problem with the official version of wxWidgets, compilation is impossible.
It is deleted the setup.h who was in "wxWidgets \ include \ wx \ msw"
So it is totally impossible to compile.

**Comment by tambry** [comment]
@Zangetsu38 Bigpet already said he was going to fix that.

**Comment by Bigpet** [comment]
@Zangetsu38 I wouldn't call it "impossible". It's completely reasonable to just introduce a dummy project in the rpcs3 solution, mark that as a dependency for the wx projects (dependencies are managed in the *.sln file so we don't need to change the wx vcxproj files). Then have that dummy project just copy a setup.h from the rpcs3 repository into the wxWidgets folder

**Comment by Zangetsu38** [comment]
@Bigpet  Yes but no, it is not a solution that, we should not have copied a file, it makes no sense, it has always worked in the not where you start compiling and it's good.
from the moment a project points to a file that does not exist, dsl to say this, but huge fail, it's not normal, it's completely illogical, that's all.

**Comment by Bigpet** [comment]
I'm sorry but copying a config file from another source does not seem like it's illogical at all.

That's what package builders for all linux distros do, that's what we essentially do with llvm_build.vcxproj when we're passing config parameters to cmake. How is this different from that?

**Comment by Zangetsu38** [comment]
@Bigpet I'm not unreasonable.
It's just pure logic, when compiling a project has not copied a file and the renamed, it does not make any sense.
And I just corrected this problem.

**Comment by danilaml** [comment]
@tambry I made a pull request to vlj branch yesterday fixing travis/linux https://github.com/vlj/rpcs3/pull/39. Waiting for merge. Also I sort of fixed tests too https://github.com/danilaml/rpcs3/commits/llvm3.6-tests

**Comment by Bigpet** [comment]
It has already been explained to you that it's not part of the repository so that you can customize your build-config and still pull updates and even make pull-requests without rebasing,cherry-picking or changing the default settings. 
People from wxWidgets themselves have explained that to you, how can you still think that you're in the right here?

Sure in an ideal world there would be a "pull file but don't have changes dirty the repo" feature in git but because there is no such thing as far as I'm aware (sparse checkouts and "update-index" all require client-side changes which makes them unpractical in this case).
This should really not be that hard to understand.

**Comment by Zangetsu38** [comment]
@Bigpet  My Pull request just fix compilation problem, currently the pull request is endomagé, through my fault, and because of a forgotten ^^

Currently this pull request is damaged because of me and because of forgetting to webview.

**Comment by Nekotekina** [comment]
Merged.

**Comment by Zangetsu38** [comment]
@Nekotekina problem, submodule wxWidgets, commit not exist 
capture d ecran 2015-07-23 10 24 38

**Comment by Bigpet** [comment]
@Zangetsu38  you need to use "submodule sync" if the repository changes

**Comment by Zangetsu38** [comment]
@Bigpet  It was enough especially agreed: https://github.com/RPCS3/wxWidgets/pull/1#event-363186046
It would have updated and it would have solved that problem.
WxWidgets yours is really old.

**Pull request closed without merging**